src/InsetOrderOptimizer.cpp

```diff
@@ -99,127 +107,234 @@ bool InsetOrderOptimizer::optimize(const WallType& wall_type)
+        const bool linked_path = p_start != p_end;
+        
+        
+        added_something = true;
+        gcode_writer.setExtruder_addPrime(storage, gcode_layer, extruder_nr);
+        gcode_layer.setIsInside(true); //Going to print walls, which are always inside.
+        gcode_layer.addWall(*path.vertices, path.start_vertex, settings, non_bridge_config, bridge_config, wipe_dist, flow, retract_before, path.is_closed, backwards, linked_path);
+        added_something = true;
+    }
     return added_something;
 }
 
-size_t InsetOrderOptimizer::getOuterRegionId(const VariableWidthPaths& toolpaths, size_t& out_max_region_id)
+
+
+
+std::unordered_set<std::pair<const ExtrusionLine*, const ExtrusionLine*>> InsetOrderOptimizer::getRegionOrder(const std::vector<const ExtrusionLine*>& input, const bool outer_to_inner)
 {
-    // Polygons show up here one by one, so there are always only a) the outer lines and b) the lines that are part of the holes.
-    // Therefore, the outer-regions' lines will always have the region-id that is larger then all of the other ones.
+    std::unordered_set<std::pair<const ExtrusionLine*, const ExtrusionLine*>> order_requirements;
 
-    // First, build the bounding boxes:
-    std::map<size_t, AABB> region_ids_to_bboxes; //Use a sorted map, ordered by region_id, so that we can find the largest region_id quickly.
-    for (const VariableWidthLines& path : toolpaths)
-    {
-        for (const ExtrusionLine& line : path)
+    coord_t max_line_w = 0u;
+    { // compute max_line_w
+        for (const ExtrusionLine* line : input)
         {
-            AABB& aabb = region_ids_to_bboxes[line.region_id]; // Empty AABBs are default initialized when region_ids are encountered for the first time.
-            for (const auto& junction : line.junctions)
+            for (const ExtrusionJunction& junction : *line)
             {
-                aabb.include(junction.p);
+                max_line_w = std::max(max_line_w, junction.w);
             }
         }
     }
-
-    // Then, the largest of these will be the one that's needed for the outer region, the others' all belong to hole regions:
-    AABB outer_bbox;
-    size_t outer_region_id = 0; // Region-ID 0 is reserved for 'None'.
-    for (const auto& region_id_bbox_pair : region_ids_to_bboxes)
+    if (max_line_w == 0u) return order_requirements;
+    
+    struct LineLoc
     {
-        if (region_id_bbox_pair.second.contains(outer_bbox))
+        ExtrusionJunction j;
+        const ExtrusionLine* line;
+    };
+    struct Locator
+    {
+        Point operator()(const LineLoc& elem)
         {
-            outer_bbox = region_id_bbox_pair.second;
-            outer_region_id = region_id_bbox_pair.first;
+            return elem.j.p;
         }
-    }
-
-    // Maximum Region-ID (using the ordering of the map)
-    out_max_region_id = region_ids_to_bboxes.empty() ? 0 : region_ids_to_bboxes.rbegin()->first;
-    return outer_region_id;
-}
+    };
+    
```

Comment:
I would add a comment here that you use a grid to find nearby points. These nearby points are most-likely neighbour-toolpaths.

Reply:
Done